tutoraspects/commands_v0.py

```diff
@@ -110,18 +110,83 @@ def dump_courses_to_clickhouse(context, options) -> None:
 # pylint: disable=line-too-long
 # Ex: tutor local do transform-tracking-logs --options "--source_provider MINIO --source_config '{\"key\": \"openedx\", \"secret\": \"h3SIhXAqDDxJAP6TcXklNxro\", \"container\": \"openedx\", \"prefix\": \"/tracking_logs\", \"host\": \"files.local.overhang.io\", \"secure\": false}' --destination_provider LRS --transformer_type xapi"
 @click.command(help="Uses event-routing-backends to replay tracking logs.")
-@click.option("--options", default="")
+@click.option(
+    '--source_provider', 
+    type=str, 
+    required=True, 
+    help="An Apache Libcloud provider. This is mandatory."
+)
+@click.option(
+    '--source_config', 
+    type=str, 
+    required=True, 
+    help="A JSON dictionary of configuration for the source provider. This is mandatory."
+)
+@click.option(
+    '--destination_provider', 
+    type=str, 
+    default="LRS", 
+    help="Either 'LRS' or an Apache Libcloud provider. The default is 'LRS'."
+)
+@click.option(
+    '--destination_config', 
+    type=str, 
+    help=(
+        "A JSON dictionary of configuration for the destination provider. "
+        "Optional if 'LRS' is used as the destination_provider."
+    )
+)
+@click.option(
+    '--transformer_type', 
+    type=click.Choice(['xapi', 'caliper']), 
+    required=True, 
+    help="The type of transformation to perform, either 'xapi' or 'caliper'. This is mandatory."
+)
+@click.option(
+    '--batch_size', 
+    type=int, 
+    default=10000, 
+    help="The batch size. The default is 10000."
+)
+@click.option(
+    '--sleep_between_batches_secs', 
+    type=float, 
+    default=10.0, 
+    help="The amount of time (in seconds) to sleep between sending batches. Default is 10.0 seconds."
+)
+@click.option(
+    '--dry_run', 
+    is_flag=True, 
+    help=(
+        "A flag to determine if this is a dry run. If present, all lines from all files "
+        "will be attempted to be transformed, but won't be sent to the destination."
+    )
+)
 @click.pass_obj
-def transform_tracking_logs(context, options) -> None:
+def transform_tracking_logs(context, **kwargs) -> None:
     """
     Job that proxies the dump_courses_to_clickhouse commands.
     """
     config = tutor_config.load(context.root)
     runner = context.job_runner(config)
 
-    command = f"""
-    ./manage.py lms transform_tracking_logs {options}
-    """
+    options = (
+        f"--source_provider {kwargs['source_provider']} "
+        f"--source_config {kwargs['source_config']} "
+        f"--destination_provider {kwargs['destination_provider']} "
+        f"--transformer_type {kwargs['transformer_type']}"
+    )    
+    
+    if 'destination_config' in kwargs and kwargs['destination_config']:
+        options += f" --destination_config {kwargs['destination_config']}"
+        
+    options += f" --batch_size {kwargs['batch_size']} --sleep_between_batches_secs {kwargs['sleep_between_batches_secs']}"
```

Comment:
I would prefer to validate every input using something like:

```python
for arg in kwargs:
    if kwargs[arg]:
        options.append(f"--{arg} {kwargs[arg]}")
```

This way, only defined values are passed for the downstream command without any intervenient from the wrapper